store/tikv/coprocessor.go

```diff
@@ -109,6 +113,19 @@ func (c *CopClient) Send(ctx context.Context, req *kv.Request, vars *kv.Variable
 	return it
 }
 
+func (c *CopClient) validateRanges(ranges []kv.KeyRange) error {
+	for i := 1; i < len(ranges); i++ {
+		if bytes.Compare(ranges[i].StartKey, ranges[i-1].EndKey) < 0 {
+			startKeyLen := len(ranges[i].StartKey)
+			if startKeyLen+1 == len(ranges[i-1].EndKey) && bytes.Equal(ranges[i].StartKey, ranges[i-1].EndKey[:startKeyLen]) && ranges[i-1].EndKey[startKeyLen] == 0 {
```

Comment:
@breeswish @zhangjinpeng1987  Please make sure this if branch is really a safe case for tikv.